### PR TITLE
RavenDB-20693 : fix possible NRE in flaky test

### PR DESCRIPTION
**Pull request by aviv86**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20693

### Type of change

- Bug fix
- Tests stabilization

### How risky is the change?

- Low

### CODE REVIEW

File: test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs [hunk 1]
```diff
@@ -308,9 +308,8 @@ public async Task CanPostAndGetDocumentsExpiration(Options options)
             }
         }
 
-        [RavenTheory(RavenTestCategory.ExpirationRefresh)]
-        [RavenData(DatabaseMode = RavenDatabaseMode.Sharded)]
-        public async Task ShouldDeleteExpiredDocumentsForSharding(Options options)
+        [RavenFact(RavenTestCategory.ExpirationRefresh)]
```
Comment (ppekrol) [inline test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs]: add also Sharding category then

File: test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs [hunk 2]
```diff
@@ -401,8 +401,10 @@ public async Task ShouldDeleteExpiredDocumentsForSharding(Options options)
                         foreach (var database in kvp.Value)
                         {
                             database.Time.UtcDateTime = () => DateTime.UtcNow.AddMinutes(10);
-                            var expiredDocumentsCleaner = database.ExpiredDocumentsCleaner;
-                            await expiredDocumentsCleaner?.CleanupExpiredDocs()!;
+                            if (database.ExpiredDocumentsCleaner == null)
+                                continue;
```
Comment (ppekrol) [inline test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs]: when is this possible? you are setting the expiration so it should not be null?

File: test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs (hunk 1 above)
Comment (aviv86) [reply test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs]: done

File: test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs (hunk 2 above)
Comment (aviv86) [reply test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs]: right, we're setting expiration so `ExpiredDocumentsCleaner` should not be null
but that's what made the test [fail](https://jenkins.hibernatingrhinos.com/job/Testsv60_Winx86/149/testReport/SlowTests.Server.Documents.Expiration/ExpirationTests/Slow_tests___ShouldDeleteExpiredDocumentsForSharding_options___DatabaseMode___Sharded___SearchEngineMode___Lucene_/)

we're setting the `ExpiredDocumentsCleaner` in [InitializeFromDatabaseRecord](https://github.com/ravendb/ravendb/blob/v6.0/src/Raven.Server/Documents/DocumentDatabase.cs#L1671) (called by `UpdateOnStateChange`), so I'm guessing that on rare occasions `UpdateOnStateChange` doesn't happen fast enough?

should I use `WaitForValue` in the test to wait for `cleaner != null` ?

Comment (ppekrol) [reply test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs]: I think it would be better to use WaitForValue, yes

Comment (aviv86) [reply test/SlowTests/Server/Documents/Expiration/ExpirationTests.cs]: done

**Pull request merged**